crates/oxide/src/main.rs

```diff
@@ -0,0 +1,65 @@
+use std::hint::black_box;
+use tailwindcss_oxide::cursor::Cursor;
+use tailwindcss_oxide::extractor::machine::{Machine, MachineState};
+use tailwindcss_oxide::extractor::{Extracted, Extractor};
+use tailwindcss_oxide::throughput::Throughput;
+
+fn run_full_extractor(input: &[u8]) -> Vec<&[u8]> {
+    Extractor::new(input)
+        .extract()
+        .into_iter()
+        .map(|x| match x {
+            Extracted::Candidate(bytes) => bytes,
+            Extracted::CssVariable(bytes) => bytes,
+        })
+        .collect::<Vec<_>>()
+}
+
+fn _run_machine<T: Machine>(input: &[u8]) -> Vec<&[u8]> {
+    let len = input.len();
+    let mut machine = T::default();
+    let mut cursor = Cursor::new(input);
+    let mut result = Vec::with_capacity(25);
+
+    while cursor.pos < len {
+        if let MachineState::Done(span) = machine.next(&mut cursor) {
+            result.push(span.slice(input));
+        }
+
+        cursor.advance();
+    }
+
+    result
+}
+
+fn run(input: &[u8]) -> Vec<&[u8]> {
+    // _run_machine::<tailwindcss_oxide::extractor::arbitrary_property_machine::ArbitraryPropertyMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::arbitrary_value_machine::ArbitraryValueMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::arbitrary_variable_machine::ArbitraryVariableMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::candidate_machine::CandidateMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::css_variable_machine::CssVariableMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::modifier_machine::ModifierMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::named_utility_machine::NamedUtilityMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::named_variant_machine::NamedVariantMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::string_machine::StringMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::utility_machine::UtilityMachine>(input)
+    // _run_machine::<tailwindcss_oxide::extractor::variant_machine::VariantMachine>(input)
```

Comment:
We can delete these if we want, but it allowed me to test an individual machine in isolation so I could look at Instruments to figure out bottlenecks.

We don't even publish this anywhere so I figured to just keep it here for future use.